### ISSUE TITLE
Layout: CSP problem while update

### ISSUE THREAD

**Issue by lih13800** [primefaces-extensions/primefaces-extensions#964]
### Describe the bug

There is Content Security Policy errors occurring if the update scope contains pe:layout component. The problem exists on version 11.0.5 and 12.0.0.

In the test sample: 
- the button "update by id" update specific id, so it works properly. 
- the button "update by form" update the form which contains the < pe:layout> have errors shown below
![2525-csp](https://user-images.githubusercontent.com/114792410/194928602-97f79c8a-8ea2-40d2-bf67-0ff6cd426fc4.png)

If the CSP missing in the web.xml, please add it:
```
<context-param>
	    <param-name>primefaces.CSP</param-name>
	    <param-value>true</param-value>
</context-param>
```
[primefaces-test-master.zip](https://github.com/primefaces-extensions/primefaces-extensions/files/9748251/primefaces-test-master.zip)

### Reproducer

1. Go to input text field to input something
2. Click on 'update by form'
3. See errors on the browser console

### Expected behavior

It should be no errors

### PrimeFaces Extensions version

12.0.0

### JSF implementation

All

### JSF version

2.2

**Comment by melloware** [comment]
As you know the layout is quite a complex JS compoennt and I never update it fully.  However I did fix your example by moving your form inside the layout.

```xml
<pe:layout>
			<h:form id="frmTest">
				<pe:layoutPane position="west">
					<p:panel>
						<p:outputLabel value="Update by Id: "/>
						<p:outputLabel id="idOutput" value="#{testView.inputString}"/>
					</p:panel>
				</pe:layoutPane>
				<pe:layoutPane position="center">
					<p:panel>
						<p:outputLabel value="Update by @form: "/>
						<p:outputLabel id="formOutput" value="#{testView.inputString}"/>
					</p:panel>
					<p:separator/>
					<p:inputText id="input" value="#{testView.inputString}"/>
					<p:commandButton for="input" value="update by id" update="idOutput"/>
					<p:commandButton for="input" value="update by form" update="@form"/>
				</pe:layoutPane>
			</h:form>
		</pe:layout>
```

**Comment by melloware** [comment]
But as a warning I would avoid updating the entire layout component.

**Comment by mihaiDumea** [comment]
Can you check if `FastStringWriter` can be wrapped with `CspResponseWriter` in [encodeBegin](https://github.com/primefaces-extensions/primefaces-extensions/blob/5b7c100c82cff0b1e85a8bbd09790a1bfecaded0/core/src/main/java/org/primefaces/extensions/component/layout/LayoutRenderer.java#L63)?

**Comment by melloware** [comment]
@mihaiDumea I was going to try this again after your latest fixes to see if it was still an issue. Did you verify whether it was or not?

**Comment by mihaiDumea** [comment]
I did try with all the fixes, it still generates inline event handlers. The HTML generation does not go through the `CspResponseWriter`.

**Comment by melloware** [comment]
AHHHH that makes sense.  I would be open to a PR.

**Comment by melloware** [comment]
I submitted this PR: https://github.com/primefaces/primefaces/pull/10659

**Comment by melloware** [comment]
Fixed with PF 13.0.2
